website/components/config-entry-reference/index.jsx

```diff
@@ -0,0 +1,265 @@
+ *
+ * Examples:
+ * - Protocol => protocol
+ * - MeshGateway => meshGateway
+ * - ACLToken => aclToken
+ * - HTTP => http
+ *
+ * @param {string} hclKey
+ * @returns {string}
+ */
+function toYAMLKeyName(hclKey) {
+  // Handle something like HTTP.
+  if (hclKey.toUpperCase() === hclKey) {
+    return hclKey.toLowerCase()
+  }
+
+  let indexFirstLowercaseChar = 0
+  for (let i = 0; i < hclKey.length; i++) {
+    if (hclKey[i].toLowerCase() === hclKey[i]) {
+      indexFirstLowercaseChar = i
+      break
+    }
+  }
+
+  // Special case to handle something like ACLToken => aclToken.
+  if (indexFirstLowercaseChar > 1) {
+    indexFirstLowercaseChar--
+  }
+
+  let yamlKey = ''
+  for (let i = 0; i < indexFirstLowercaseChar; i++) {
+    yamlKey += hclKey[i].toLowerCase()
+  }
+  yamlKey += hclKey.split('').slice(indexFirstLowercaseChar).join('')
+
+  return yamlKey
+}
+
+/**
+ * Converts a markdown string to its HTML representation.
+ * Currently it only supports inline code blocks (e.g. `code here`) and
+ * links (e.g. [link text](http://link-url) because these were the most
+ * commonly used markdown features in the key descriptions.
+ *
+ * @param {string} markdown the input markdown
+ * @returns {string}
+ */
+function markdownToHtml(markdown) {
+  let html = markdown
+
+  // Replace inline code blocks defined by backticks with <code></code>.
+  while (html.indexOf('`') > 0) {
+    html = html.replace('`', '<code>')
+    if (html.indexOf('`') <= 0) {
+      throw new Error(`'${markdown} does not have matching '\`' characters`)
+    }
+    html = html.replace('`', '</code>')
+  }
+
+  // Replace links, e.g. [link text](http://link-url),
+  // with <a href="http://link-url">link text</a>.
+  return html.replace(/\[(.*?)]\((.*?)\)/g, '<a href="$2">$1</a>')
+}
+
+/**
+ * Returns true if key is a key used at the top level of a CRD. By top level we
+ * mean not nested under any other key.
+ *
+ * @param {string} name name of the key
+ *
+ * @return {boolean}
+ */
+function isTopLevelKubeKey(name) {
+  return (
+    name.toLowerCase() === 'metadata' ||
+    name.toLowerCase() === 'kind' ||
+    name.toLowerCase() === 'apiversion'
+  )
+}
+
+// Copied from https://github.com/hashicorp/nextjs-scripts/blob/04917da2191910d490182250d2828372aa1221c0/lib/providers/docs/index.jsx
+// because there's no way to import it right now.
+function Tabs({ children }) {
+  if (!Array.isArray(children))
+    throw new Error('Multiple <Tab> elements required')
+
+  return (
+    <span className={s.tabsRoot}>
+      <TabsBase
+        items={children.map((Block) => ({
+          heading: Block.props.heading,
+          // eslint-disable-next-line react/display-name
+          tabChildren: () => Block,
+        }))}
+      />
+    </span>
+  )
+}
+
+// Copied from https://github.com/hashicorp/nextjs-scripts/blob/04917da2191910d490182250d2828372aa1221c0/lib/providers/docs/index.jsx
+// because there's no way to import it right now.
```

Comment:
We could export these it wouldn't be super difficult - want me to put up a PR for this? It would reduce debt in this PR I think

Reply:
Yeah I think that's the best path if it's doable. I'd probably want to merge this first unless that PR was incoming soon though.

Reply:
I can land that one today, i'll get something going shortly